draft-ietf-masque-h3-datagram.md

```diff
@@ -230,9 +227,20 @@ Capsule Data:
 
 : Data whose semantics depends on the Capsule Type.
 
+Unless otherwise specified, all Capsule Types are defined as opaque to
+intermediaries. Intermediaries MUST forward all received opaque CAPSULE frames
+in their unmodified entirety. Intermediaries MUST NOT send any opaque CAPSULE
+frames other than the ones it is forwarding. All Capsule Types defined in this
+document are opaque, with the exception of the DATAGRAM Capsule, see
+{{datagram-capsule}}. Definitions of new Capsule Types MAY specify that the
+newly introduced type is not opaque. Intermediaries MUST treat unknown Capsule
+Types as opaque.
+
+Intermediaries respect the order of CAPSULE frames: if an intermediary receives
+two CAPSULE frames in a given order, it MUST forward them in the same order.
+
```

Comment:
Order of transparent CAPSULEs (such as DATAGRAM) is no longer guaranteed, because an H3 -> H2 intermediary can have reordering effects.

Reply:
Good catch, fixed